application/models/Template.php

```diff
@@ -365,6 +365,12 @@ public static function getTemplatePath($sTemplateName = "")
      */
     public static function getTemplateConfiguration($sTemplateName = null, $iSurveyId = null, $iSurveyGroupId = null, $bForceXML = false, $abstractInstance = false)
     {
+        $event = new PluginEvent('beforeGetTemplate');
+        $event->set('surveyId', $iSurveyId);
+        $event->set('template', $sTemplateName);
+        $event = App()->getPluginManager()->dispatchEvent($event);
+        $sTemplateName = $event->get('template');
+
```

Comment:
getTemplateConfiguration already must be improved … 

Working at https://github.com/LimeSurvey/LimeSurvey/commit/c6bed7623f81f7c01d03c3c5adc8fa6bd8ea7a59 i see we need more optimization on TemplateCOnfiguration system.

Here i think it's more in https://github.com/LimeSurvey/LimeSurvey/blob/9b6a908112a734612a5655424e874d207d63dfae/application/models/Template.php#L550 you need an event ?